go/tasks/pluginmachinery/flytek8s/container_helper.go

```diff
@@ -303,9 +310,7 @@ func AddFlyteCustomizationsToContainer(ctx context.Context, parameters template.
 			" Resources [%v] with mode [%v]", res, platformResources, container.Resources, mode)
 
 		switch mode {
-		case ResourceCustomizationModeAssignResources:
-			container.Resources = ApplyResourceOverrides(*res, *platformResources, assignIfUnset)
-		case ResourceCustomizationModeMergeExistingResources:
+		case ResourceCustomizationModeAssignResources, ResourceCustomizationModeMergeExistingResources:
```

Comment:
I don't think that removing support for `ResourceCustomizationModeAssignResources` is the right call here. Could we just update the call to `AddFlyteCustomizationsToContainer` in the [ToK8sContainer function](https://github.com/flyteorg/flyteplugins/blob/18a594ecef8bf744b42d9a824df2d53635aa31d2/go/tasks/pluginmachinery/flytek8s/container_helper.go#L254) to use `ResourceCustomizationModeMergeExistingResources` instead? IIUC this will achieve the same thing, but please correct me.